packages/api/src/brevo-contact/brevo-contact.resolver.ts

```diff
@@ -184,13 +210,103 @@ export function createBrevoContactResolver({
+                }
+
+                return SubscribeResponse.ERROR_UNKNOWN;
+            }
+        }
+
         @Mutation(() => Boolean)
         @AffectedEntity(BrevoContact)
         async deleteBrevoContact(
             @Args("id", { type: () => Int }) id: number,
             @Args("scope", { type: () => Scope }, new DynamicDtoValidationPipe(Scope)) scope: typeof Scope,
         ): Promise<boolean> {
-            return this.brevoContactsApiService.deleteContact(id, scope);
+            const contact = await this.brevoContactsApiService.findContact(id, scope);
+            const where: FilterQuery<TargetGroupInterface> = { scope, isMainList: false, isTestList: true };
+            const testTargetGroup = await this.targetGroupRepository.findOne(where);
+            const contactIncludesTestList = testTargetGroup?.brevoId ? contact.listIds.includes(testTargetGroup.brevoId) : false;
+
+            if (testTargetGroup && contactIncludesTestList) {
+                const testListId = testTargetGroup.brevoId;
+
+                const unlinkListIds = contact.listIds.filter((id) => id !== testListId);
+
+                await this.brevoContactsApiService.updateContact(
+                    contact.id,
+                    {
+                        listIds: [testListId],
+                        unlinkListIds,
+                    },
+                    scope,
+                );
+                return true;
+            } else {
+                return this.brevoContactsApiService.deleteContact(id, scope);
+            }
+        }
+
+        @Mutation(() => Boolean)
+        @AffectedEntity(BrevoContact)
+        async deleteBrevoTestContact(
+            @Args("id", { type: () => Int }) id: number,
+            @Args("scope", { type: () => Scope }, new DynamicDtoValidationPipe(Scope)) scope: typeof Scope,
+        ): Promise<boolean> {
+            const contact = await this.brevoContactsApiService.findContact(id, scope);
+            const where: FilterQuery<TargetGroupInterface> = { scope, isMainList: false, isTestList: true };
+            const testTargetGroup = await this.targetGroupRepository.findOne(where);
+            const contactIncludesTestList = testTargetGroup?.brevoId ? contact.listIds.includes(testTargetGroup.brevoId) : false;
+
+            if (testTargetGroup && contactIncludesTestList) {
+                const testListId = testTargetGroup.brevoId;
+
+                const linkListIds = contact.listIds.filter((id) => id !== testListId);
+
+                await this.brevoContactsApiService.updateContact(
+                    contact.id,
+                    {
+                        listIds: linkListIds,
+                        unlinkListIds: [testListId],
+                    },
+                    scope,
+                );
+                return true;
+            } else {
+                return this.brevoContactsApiService.deleteContact(id, scope);
+            }
```

Comment:
What exactly should happen here?

**Am I right with that:**
When the contact is only in the testlist, and nowhere else, it should be deleted.
Otherwise it should just be unlinked

**Currently it works like this:**
 - When there is a testTargetGroup AND the contact is in the list, it will be unlinked, otherwise it will be deleted? Does this make sense?

Reply:
Yes, you are absolutely right! 
Now I check, if the contact is in the MainList and if he is, the test list is unlinked. Otherwise deleted